lib/Basic/SourceManager.cpp

```diff
@@ -140,6 +140,13 @@ char const *glu::SourceManager::getCharacterData(SourceLocation loc) const
     return buffer.data() + offsetInFile;
 }
 
+llvm::StringRef
+glu::SourceManager::getCharacterDataInStringRef(SourceLocation loc) const
+{
+    auto *data = getCharacterData(loc);
+    return data ? llvm::StringRef(data) : llvm::StringRef();
```

Comment:
No, this will create a string ref from a null terminated big buffer which is extremely inefficient and completely useless. Keep the pointer.